File: 1-two-sum/1-two-sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    
    for i, j := range nums{
        if j, ok := m[target - j]; ok{
            return []int{j, i}
        }
        m[j] = i
    }
    return []int{}
}

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> m;
        int k;
        for(int i = 0; i < nums.size(); i++){
            k = target - nums[i];
            if (m.find(k) != m.end()){
                return {i,m[target - nums[i]]};
            }
            m[nums[i]] = i;
        }
        return {};
    }
};